### ISSUE TITLE
[Documentation] Descriptions for built-ins

### ISSUE THREAD

**Issue by gregsdennis** [json-e/json-e#520]
The current docs for [built-ins](https://json-e.js.org/built-ins.html) is a bit sparse and could be improved.

The examples get you a long way, but there are behaviors that don't come out with what's there.

For example, I recently had to update my implementation to support returning the input string for `split()` when the second parameter was missing.

Without a description of the function, this kind of behavior is hard to identify.

**Comment by djmitche** [comment]
Agreed, each section should describe each of the provided built-ins using a sentence or two.
